adapters/LiftoffMonetize/LiftoffMonetizeAdapter/Bidding/GADMediationVungleBanner.m

```diff
@@ -101,154 +98,89 @@ - (void)requestBannerAd {
+- (void)bannerAdDidLoad:(VungleBanner *)banner {
   _bannerView = [[UIView alloc]
       initWithFrame:CGRectMake(0, 0, _bannerSize.size.width, _bannerSize.size.height)];
-  self.bannerState = BannerRouterDelegateStateWillPlay;
-  NSError *error = [self renderAd];
-  if (error) {
-    _adLoadCompletionHandler(nil, error);
-    return;
-  }
-
   if (_adLoadCompletionHandler) {
+    [_bannerAd presentOn:_bannerView];
     _delegate = _adLoadCompletionHandler(self, nil);
   }
+}
 
-  if (!_delegate) {
-    [[GADMAdapterVungleBiddingRouter sharedInstance] removeDelegate:self];
-    return;
-  }
+- (void)bannerAdDidFailToLoad:(VungleBanner *)banner withError:(NSError *)error {
+  NSError *gadError = GADMAdapterVungleErrorToGADError(GADMAdapterVungleErrorAdNotPlayable,
+                                                       error.code, error.localizedDescription);
+  _adLoadCompletionHandler(nil, gadError);
 }
 
-- (void)adNotAvailable:(nonnull NSError *)error {
-  if (_isAdLoaded) {
-    // Already invoked an ad load callback.
-    return;
-  }
-  _adLoadCompletionHandler(nil, error);
+- (void)bannerAdWillPresent:(VungleBanner *)banner {
+  // Google Mobile Ads SDK doesn't have a matching event.
 }
 
-- (void)willShowAd {
-  self.bannerState = BannerRouterDelegateStatePlaying;
+- (void)bannerAdDidPresent:(VungleBanner *)banner {
+  // Google Mobile Ads SDK doesn't have a matching event.
 }
 
-- (void)didViewAd {
-  [_delegate reportImpression];
+- (void)bannerAdDidFailToPresent:(VungleBanner *)banner withError:(NSError *)error {
+  NSError *gadError = GADMAdapterVungleErrorToGADError(GADMAdapterVungleErrorRenderBannerAd,
+                                                       error.code, error.localizedDescription);
+  [_delegate didFailToPresentWithError:gadError];
 }
 
-- (void)willCloseAd {
-  self.bannerState = BannerRouterDelegateStateClosing;
+- (void)bannerAdWillClose:(VungleBanner *)banner {
   // This callback is fired when the banner itself is destroyed/removed, not when the user returns
   // to the app screen after clicking on an ad. Do not map to adViewWillDismissScreen:.
 }
 
-- (void)didCloseAd {
-  self.bannerState = BannerRouterDelegateStateClosed;
+- (void)bannerAdDidClose:(VungleBanner *)banner {
   // This callback is fired when the banner itself is destroyed/removed, not when the user returns
   // to the app screen after clicking on an ad. Do not map to adViewDidDismissScreen:.
 }
 
-- (void)trackClick {
+- (void)bannerAdDidTrackImpression:(VungleBanner *)banner {
+  [_delegate reportImpression];
+}
+
+- (void)bannerAdDidClick:(VungleBanner *)banner {
   [_delegate reportClick];
 }
 
-- (void)willLeaveApplication {
+- (void)bannerAdWillLeaveApplication:(VungleBanner *)banner {
   [_delegate willBackgroundApplication];
 }
 
-- (void)rewardUser {
-  // Do nothing.
-}
+#pragma mark - GADMAdapterVungleDelegate
 
-- (void)didShowAd {
-  // Do nothing.
+- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
```

Comment:
Can you remind me again when would/should this `initialized` callback be called? Do we really need this callback?

Reply:
This `initialized` callback is called if the Vungle SDK wasn't initialized yet. It is triggered as a form of lazy initialization. If the publisher's app initializes the Vungle SDK before the load request, it will never be hit. If the Vungle SDK hasn't been initialized, the load request will first attempt initialization and on the callback, it will trigger this.